src/main/resources/public/js/recipes.js

```diff
@@ -0,0 +1,282 @@
+    recipeList.forEach((recipe, idx, arr) => {
+      $('#recipe-list').append(
+        `<button class="recipe-list list-group-item list-group-item-action " type="button" data-id="${
+          recipe.id}">${recipe.name}</button>`,
+      );
+    });
+    addRecHandler($('button.recipe-list'));
+  }, 'json');
+  currentRec = null;
+  currentIdx = null;
+}
+
+$(document).ready(() => {
+  virtualRestart(null, null);
+  $('#recipe-details-container').append(createIngDiv(''));
+  $('#addIng').click(() => {
+    ingApply('');
+  });
+  $('#cancelIng').click(() => {
+    ingCancel('');
+  });
+  $('#new-recipe-name').keypress((event) => {
+    if (event.keyCode === 13) {
+      $('#addRec').click();
+    }
+  });
+  $('#renameRecInput').keypress((event) => {
+    if (event.keyCode === 13) {
+      $('#applyRecRename').click();
+    }
+  });
+  $('input.inputIngInfo').keypress(function (event) {
+    const idx = $(this).data('idx');
+    if ($(`#unit${idx}`).val() !== '') {
+      if ($(`#quantity${idx}`).val() !== '') {
+        if ($(`#name${idx}`).val() !== '') {
+          if (event.keyCode === 13) {
+            $(`#addIng${idx}`).click();
+          }
+        }
+      }
+    }
+  });
+
+  $('#plusRec').click(() => {
+    $('#new-recipe-name').val('');
+    $('#recDiv').show();
+    $('#new-recipe-name').focus();
+  });
+  $('#addRec').click(() => {
+    $('#recDiv').hide();
+    const recipe = { name: $('#new-recipe-name').val(), ingredients: [] };
+    $.post({
+      url: 'recipes',
+      data: JSON.stringify(recipe),
+      contentType: 'application/json',
+      dataType: 'json',
+      success(data, status) {
+        $('#recipe-list').append(
+          `<button class="recipe-list list-group-item list-group-item-action " type="button" data-id="${
+            data.id}">${data.name}</button>`,
+        );
+        const lastButton = $('button.recipe-list.list-group-item-action:last-child');
+        addRecHandler(lastButton);
+        lastButton.trigger('click');
+      },
+    });
+  });
+  $('#plusIng').click(() => {
+    $('#unit').val('');
+    $('#name').val('');
+    $('#quantity').val('');
+    $('#ingDiv').show();
+  });
+  $('#delRec').click(() => {
+    $.ajax({
+      type: 'DELETE',
+      url: `/recipes/${currentRec.id}`,
+      dataType: 'json',
+      statusCode: {
+        200: virtualRestart,
+      },
+    });
+  });
+  $('#editRecName').click(() => {
+    $('#recTitle').hide();
+    $('#editRecName').hide();
+    $('#renamed-recipe-name').show();
+    $('#renameRecInput').val($('#recTitle').text());
+  });
+  $('#applyRecRename').click(() => {
+    currentRec.name = $('#renameRecInput').val();
+    $.post({
+      url: 'recipes',
+      data: JSON.stringify(currentRec),
+      contentType: 'application/json',
+      dataType: 'json',
+      success(data, status) {
+        $('button.recipe-list.active').text(data.name);
+      },
+    });
+    $('#recTitle').show();
+    $('#editRecName').show();
```

Comment:
If you never hide it, there's no reason to explicitly show it (it'll be shown by default). So you can remove this line.